File: Programming Language(ver.2)/C#/310.cs
using System;

public class TQC310
{
    public static void Main(string[] args)
    {
        int number;
        number = int.Parse(Console.ReadLine());

        Console.WriteLine(compute(number));
    }

    public static int compute(int number) {
        int sum = 0;
        for (int i = 1; i < number; i++) {
            if (IsArmstrongNumber(i)) {
                sum += i;
                Console.WriteLine(i);
            }
        }
        return sum;
    }

    public static bool IsArmstrongNumber(int number) {
        int sum = 0;
        string numberString = number.ToString();
        int k = numberString.Length;
        for (int i = 0; i < k; i++) {
            sum += (int)(Math.Pow(numberString[i] - '0', k));
        }
        
        return sum == number;
    }
}

File: Programming Language(ver.2)/Go/310.go
package main
import (
	"fmt"
	"math"
	"strconv"
)

func compute(n int) int {
	total := 0
	for i := 1; i < n; i++{
		s := strconv.Itoa(i)
		sum := 0
		for j := 0; j < len(s); j++{
			digit := float64(int(s[j] - '0'))
			length := float64(len(s))
			//fmt.Println(i, digit, length)
			sum += int(math.Pow(digit, length))
		}
		
		if sum == i{
			total += i
			fmt.Println(i)
		}
	}
	return total
}

func main(){
	var n int
	fmt.Scanln(&n)
	fmt.Println(compute(n))
}
